src/main/java/org/broadinstitute/dsm/util/ElasticSearchUtil.java

```diff
@@ -382,6 +382,33 @@ public static void writeWorkflow(@NonNull WorkflowForES workflowForES, boolean c
         }
     }
 
+    public static void removeWorkflowIfNoDataOrWrongSubject(RestHighLevelClient client, String ddpParticipantId, DDPInstance ddpInstance, String collaboratorParticipantId) {
+        String index = ddpInstance.getParticipantIndexES();
+        try {
+            Map<String, Object> workflowMapES = getObjectsMapHavingClient(client, index, ddpParticipantId, ESObjectConstants.WORKFLOWS);
+            if (workflowMapES == null || workflowMapES.isEmpty()) {
+                return;
+            }
+            List<Map<String, Object>> workflowListES = (List<Map<String, Object>>) workflowMapES.get(ESObjectConstants.WORKFLOWS);
+            int initialListSize = workflowListES.size();
+            workflowListES.removeIf(workflow -> !workflow.containsKey("data") ||
+                    (((Map<String, Object>) workflow.get("data")).get("subjectId") != null
+                            && ((Map<String, Object>) workflow.get("data")).get("subjectId").equals(collaboratorParticipantId)));
```

Comment:
Can you yoda-ify this to avoid an NPE so it's

`collaboratorParticipantId.equals(((Map)workflow.get("data")).get("subjectId"))` ?